src/backend/plugins/config-xdp/config-xdp.c

```diff
@@ -0,0 +1,164 @@
+
+enum {
+  PROP_0,
+  PROP_CONFIG_OPTION
+};
+
+static void
+px_config_xdp_init (PxConfigXdp *self)
+{
+  g_autoptr (GDBusConnection) connection = NULL;
+  g_autoptr (GError) error = NULL;
+  g_autofree char *path = g_build_filename (g_get_user_runtime_dir (), "flatpak-info", NULL);
+
+  self->available = FALSE;
+  if (!g_file_test (path, G_FILE_TEST_EXISTS))
+    return;
+
+  connection = g_bus_get_sync (G_BUS_TYPE_SESSION, NULL, &error);
+  if (error) {
+    g_warning ("Could not access dbus session: %s", error->message);
+    return;
+  }
+
+  self->proxy_resolver = g_dbus_proxy_new_sync (connection, G_DBUS_PROXY_FLAGS_NONE, NULL, "org.freedesktop.portal.Desktop", "/org/freedesktop/portal/desktop", "org.freedesktop.portal.ProxyResolver", NULL, &error);
+  if (error) {
+    g_warning ("Could not access proxy resolver: %s", error->message);
+    return;
+  }
+
+  self->available = TRUE;
+}
+
+static void
+px_config_xdp_dispose (GObject *object)
+{
+  PxConfigXdp *self = PX_CONFIG_XDP (object);
+
+  g_clear_object (&self->proxy_resolver);
+}
+
+static void
+px_config_xdp_set_property (GObject      *object,
+                            guint         prop_id,
+                            const GValue *value,
+                            GParamSpec   *pspec)
+{
+  switch (prop_id) {
+    case PROP_CONFIG_OPTION:
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+px_config_xdp_get_property (GObject    *object,
+                            guint       prop_id,
+                            GValue     *value,
+                            GParamSpec *pspec)
+{
+  switch (prop_id) {
+    case PROP_CONFIG_OPTION:
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+px_config_xdp_class_init (PxConfigXdpClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->dispose = px_config_xdp_dispose;
+  object_class->set_property = px_config_xdp_set_property;
+  object_class->get_property = px_config_xdp_get_property;
+
+  g_object_class_override_property (object_class, PROP_CONFIG_OPTION, "config-option");
+}
+
+static void
+px_config_xdp_get_config (PxConfig     *config,
+                          GUri         *uri,
+                          GStrvBuilder *builder)
+{
+  g_autoptr (GVariant) var = NULL;
+  g_autoptr (GError) error = NULL;
+  g_autoptr (GVariantIter) iter = NULL;
+  PxConfigXdp *self = PX_CONFIG_XDP (config);
+  g_autofree char *uri_str = NULL;
+  const char *str;
+
+  if (!self->available)
+    return;
+
+  uri_str = g_uri_to_string (uri);
+  var = g_dbus_proxy_call_sync (self->proxy_resolver, "Lookup", g_variant_new ("(s)", uri_str), 0, -1, NULL, &error);
+  if (error) {
+    g_warning ("Could not query proxy: %s", error->message);
+    return;
+  }
+
+  if (!var) {
+    g_warning ("Failed to find proxy");
+    return;
+  }
```

Comment:
The API contract here is that if `var` is NULL, then `error` is set, so this is unreachable.

Reply:
Missed it, thanks. Updated.